dashboard/app/jobs.go

```diff
@@ -455,6 +546,82 @@ func createJobResp(c context.Context, job *Job, jobKey *db.Key) (*dashapi.JobPol
 	return resp, false, nil
 }
 
+// It would be easier to just check if the User field is empty, but let's also not
+// miss the situation when some actual user sends a patch testing request without
+// patch.
+func isRetestReproJob(job *Job, build *Build) bool {
+	if job.Type != JobTestPatch || job.Patch != 0 {
+		return false
+	}
+	if job.KernelRepo != build.KernelRepo ||
+		job.KernelBranch != build.KernelBranch {
+		return false
+	}
+	return true
+}
+
+func handleRetestedRepro(c context.Context, now time.Time, job *Job, jobKey *db.Key, crashTitle string) error {
+	bugKey := jobKey.Parent()
+	crashKey := db.NewKey(c, "Crash", "", job.CrashID, bugKey)
+	crash := new(Crash)
+	if err := db.Get(c, crashKey, crash); err != nil {
+		return fmt.Errorf("failed to get crash: %v", crashKey)
+	}
+	build, err := loadBuild(c, job.Namespace, crash.BuildID)
+	if err != nil {
+		return fmt.Errorf("failed to query build: %w", err)
+	}
+	if !isRetestReproJob(job, build) {
+		return nil
+	}
+	bug := new(Bug)
+	if err := db.Get(c, bugKey, bug); err != nil {
+		return fmt.Errorf("failed to get bug: %v", bugKey)
+	}
+	// Update the crash.
+	crash.LastStaleCheck = now
+	if crashTitle == "" {
+		// If there was any crash at all, the repro is still not worth discarding.
+		crash.StaleRepro = true
+	}
+	crash.UpdateReportingPriority(build, bug)
+	if _, err := db.Put(c, crashKey, crash); err != nil {
+		return fmt.Errorf("failed to put crash: %v", err)
+	}
+	reproCrashes, crashKeys, err := queryCrashesForBug(c, bugKey, 2)
```

Comment:
Why do we need to select 2 crashes?

Reply:
Because it's possible that we've reduced the priority of the top-priority crash a few lines above. Datastore doesn't take that into account in transactions, so we query two and adjust the output.

Reply:
Though this querying won't be needed anyway if we stop recalculating ReproLevel for a bug.